data-raw/ictr/ictr_main.Rmd

```diff
@@ -0,0 +1,293 @@
+  select(-n_unit) |>
+  rename("score" = "score_unit")
+
+# scores in comparison to input products with same input sector
+scores_sector <- companies_scores |>
+  group_by(company_id, score_sector) |>
+  summarise(n_sector = n()) |>
+  mutate(share_sector = n_sector / sum(n_sector)) |>
+  select(-n_sector) |>
+  rename("score" = "score_sector")
+
+# scores in comparison to input products with same unit and input sector
+scores_unit_sec <- companies_scores |>
+  group_by(company_id, score_unit_sec) |>
+  summarise(n_unit_sec = n()) |>
+  mutate(share_unit_sec = n_unit_sec / sum(n_unit_sec)) |>
+  select(-n_unit_sec) |>
+  rename("score" = "score_unit_sec")
+```
+## 3.3 Create final dataset
+
+The final dataset should have three rows per company, i.e. one for each
+risk_group (low, medium, high) and one column each indicating the share of
+products in the respective risk groups for the four  different benchmarks (all
+input products, input products with same unit, input products with same input
+sector, and input products with same unit and input sector)
+
+```{r}
+# create dataset sceleton
+dt_sceleton <- tibble(
+  company_id = rep(unique(companies_scores$company_id), each = 3),
+  score = rep(c("high", "medium", "low"), 5),
+)
+
+# join scores with dt_sceleton so that each company is shown with 3 rows for
+# low, medium, and high, even if the share is 0.
+pctr_output <- dt_sceleton |>
+  left_join(scores_all, by = c("company_id", "score")) |>
+  left_join(scores_unit, by = c("company_id", "score")) |>
+  left_join(scores_sector, by = c("company_id", "score")) |>
+  left_join(scores_unit_sec, by = c("company_id", "score"))
+
+# replace NAs with 0
+pctr_output <- pctr_output |>
+  replace(is.na(pctr_output), 0)
+
+# rm(companies_scores)
+# rm(scores_all)
+# rm(scores_unit)
+# rm(scores_unit_sec)
+# rm(dt_sceleton)
+```
+
+## 3.4 Vizualisation example
+
+Let us plot the result for the company id
+"fleischerei-stiefsohn_00000005219477-001". The score aggregated is the
+percentage of products of that company that are based on their carbon footprint
+in either low, medium or high transition risk, relatively 4 different benchmarks
+(all input products, input products with same unit, input products with same
+input sector, and input products with same unit and input sector). To do: Create
+a function that shows the scores per benchmark for one example company, as Linda
+did in her pstr mvp.
+
+```{r}
+pstr_plot_company <- function(data, company_id) {
+  level_order <- c("low", "medium", "high")
+
+  data |>
+    filter(.data$company_id == .env$company_id) |>
+    ggplot(aes(x = .data$score, y = .data$share_all)) +
+    scale_x_discrete(limits = level_order) +
+    geom_col()
+}
+# plot for `share_all` column only (plot function will be updated later)
+pstr_plot_company(pctr_output, "fleischerei-stiefsohn_00000005219477-001")
+```
+
+## 3.5 write csv into output folder
+
+```{r}
+write_csv(pctr_output, here("03_output_file", "mvp_pctr_output.csv"))
+```
```

Comment:
In general we don't need to store the output because it can be reproduced with the code above. Just print the result as a tibble so we can see its structure.